WordPress/gutenberg
# Add success and error snackbars to the templates list page

## Description

Part of https://github.com/WordPress/gutenberg/issues/36597.

Add success and error snackbars to the templates list page. It will now:

1. Show a success notice when a template(-part) is successfully created or removed.
2. Show an error notice when there's something wrong in (1).

Since that the snackbars have similar background as the navigation panel, they will be pushed right a little bit when the panel is opened.

## How has this been tested?



1. Activate `tt1-blocks` theme
2. Go to Appearance -> Editor -> top left corner W menu -> Templates / Template Parts
3. Try creating a template or a template part
4. You won't notice the snackbar if the navigation is very fast, but with a poor network you will notice something like "Template created. Navigating to the editor..."
5. Try removing a template or a template part
6. The snackbar will show "{template title} removed"
7. If something went wrong when retrying (3) and (5) (like adding intentional `throw`), you should notice error snackbars.

## Screenshots 
https://user-images.githubusercontent.com/7753001/143198279-021d1e9f-376f-4d5f-88a8-f4ceb96015c9.mp4

## Types of changes




New feature

## Checklist:
